File: jphp-gui-ext/src/main/java/org/develnext/jphp/ext/javafx/classes/event/UXScrollEvent.java
package org.develnext.jphp.ext.javafx.classes.event;

import javafx.scene.input.ScrollEvent;
import org.develnext.jphp.ext.javafx.JavaFXExtension;
import php.runtime.annotation.Reflection.Getter;
import php.runtime.annotation.Reflection.Name;
import php.runtime.annotation.Reflection.Property;
import php.runtime.env.Environment;
import php.runtime.reflection.ClassEntity;

@Name(JavaFXExtension.NS + "event\\UXScrollEvent")
public class UXScrollEvent extends UXEvent {
    interface WrappedInterface {
        @Property double deltaX();
        @Property double deltaY();
        @Property double textDeltaX();
        @Property double textDeltaY();
        @Property double totalDeltaX();
        @Property double totalDeltaY();
        @Property double multiplierX();
        @Property double multiplierY();
        @Property int touchCount();
    }


    public UXScrollEvent(Environment env, ScrollEvent wrappedObject) {
        super(env, wrappedObject);
    }

    public UXScrollEvent(Environment env, ClassEntity clazz) {
        super(env, clazz);
    }

    @Override
    public ScrollEvent getWrappedObject() {
        return (ScrollEvent) super.getWrappedObject();
    }

    @Getter
    protected String getEventType() {
        return getWrappedObject().getEventType().getName();
    }
}

File: jphp-gui-ext/src/main/resources/JPHP-INF/sdk/php/gui/event/UXScrollEvent.php
<?php
namespace php\gui\event;

/**
 * Class UXScrollEvent
 * @package php\gui\event
 * @packages gui, javafx
 */
class UXScrollEvent extends UXEvent
{
    /**
     * @readonly
     * @var double
     */
    public $deltaX, $deltaY;

    /**
     * @readonly
     * @var double
     */
    public $textDeltaX, $textDeltaY;

    /**
     * @readonly
     * @var double
     */
    public $totalDeltaX, $totalDeltaY;

    /**
     * @readonly
     * @var double
     */
    public $multiplierX, $multiplierY;

    /**
     * @readonly
     * @var int
     */
    public $touchCount;

    /**
     * @readonly
     * @var string SCROLL, SCROLL_STARTED, SCROLL_FINISHED
     */
    public $eventType;
}